### ISSUE TITLE
How does --num_vectors in textual inversion actually work?

### ISSUE THREAD

**Issue by hkazami** [huggingface/diffusers#4151]
Hi all,

I'm currently using `examples/textual_inversion/textual_inversion.py` to learn some specific subjects, and I encountered two difficulties and would like to ask for help or ideas:

1. When using different values for the argument `--num_vectors` (all others stay the same), I observed that the loss curve of the training almost doesn't change, although the validation images are different.
2. The strange behavior above urges me to look into the code, and I found out that although the argument `--num_vectors` does create the correct number of trainable embedding vectors (corresponding to the tokens "placeholder_token
, "placeholder_token_1", ..., "placeholder_token_n"), only the 0th token ("placeholder_token") is used as prompt while training, because the train_dataset takes `args.placeholder_token` as input, instead of `placeholder_tokens`:
```
    train_dataset = TextualInversionDataset(
        ...,
        placeholder_token=args.placeholder_token,
        ...
    )
```

It seems like all other tokens are trainable but untrained? How does `--num_vectors` really work and how does it influence the training loss? Thanks for any help in advance!

Steven

**Comment by patrickvonplaten** [comment]
Hey @hkazami,

When you specify `--num_vectors` you give the textual inversion weights more capacity - i.e. more parameter they can learn. This means that you should be able to learn more complex concepts. Practically these vectors can then used just like before by the user.

**Comment by manosplitsis** [comment]
I actually tested it, and it turned out that the extra embeddings were not being trained with the current script, so I created this pull request #4452

**Comment by patrickvonplaten** [comment]
You're 100% right - great fix @manosplitsis !
